app/models/manager_refresh/inventory_collection/serialization.rb

```diff
@@ -0,0 +1,130 @@
+      def hash_to_lazy_relation(value, available_inventory_collections, depth = 0)
+        inventory_collection = available_inventory_collections[value['inventory_collection_name'].try(:to_sym)]
+        raise "Couldn't build lazy_link #{value} the inventory_collection_name was not found" if inventory_collection.blank?
+
+        inventory_collection.lazy_find(
+          hash_to_data(value['reference'], available_inventory_collections, depth + 1).symbolize_keys!,
+          :ref     => value['ref'].try(:to_sym),
+          :key     => value['key'].try(:to_sym),
+          :default => value['default']
+        )
+      end
+
+      # Converts Hash to attributes usable for building InventoryObject
+      #
+      # @param hash [Hash] Serialized InventoryObject data
+      # @param available_inventory_collections [Array<ManagerRefresh::InventoryCollection>] List of available
+      #        InventoryCollection objects
+      # @param depth [Integer] Depth of nesting for nested lazy link
+      # @return [Hash] Hash with data usable for building InventoryObject
+      def hash_to_data(hash, available_inventory_collections, depth = 0)
+        raise "Nested lazy_relation of #{inventory_collection} is too deep, left processing: #{hash}" if depth > 20
+
+        hash.transform_values do |value|
+          if value.kind_of?(Hash) && value['type'] == "ManagerRefresh::InventoryObjectLazy"
+            hash_to_lazy_relation(value, available_inventory_collections, depth)
+          elsif value.kind_of?(Array) && value.first.kind_of?(Hash) && value.first['type'] == "ManagerRefresh::InventoryObjectLazy"
+            # TODO(lsmola) do we need to compact it sooner? What if first element is nil? On the other hand, we want to
+            # deprecate Vmthis HABTM assignment because it's not effective
+            value.compact.map { |x| hash_to_lazy_relation(x, available_inventory_collections, depth) }
+          else
+            value
+          end
+        end
+      end
+
+      # Transforms data of the InventoryObject or Reference to InventoryObject into Hash
+      #
+      # @param data [Hash] Data of the InventoryObject or Reference to InventoryObject
+      # @param depth [Integer] Depth of nesting for nested lazy link
+      # @return [Hash] Serialized InventoryObject or Reference data into Hash
+      def data_to_hash(data, depth = 0)
+        raise "Nested lazy_relation of #{inventory_collection} is too deep, left processing: #{data}" if depth > 20
+
+        data.transform_values do |value|
+          if inventory_object_lazy?(value)
+            lazy_relation_to_hash(value, depth)
+          elsif value.kind_of?(Array) && (inventory_object_lazy?(value.compact.first) || inventory_object?(value.compact.first))
+            value.compact.map { |x| lazy_relation_to_hash(x, depth) }
+          elsif inventory_object?(value)
+            lazy_relation_to_hash(value, depth)
```

Comment:
Should this case be merged with https://github.com/ManageIQ/manageiq/pull/17361/files#diff-7a3a5be928f76ef87ac9cee741b1ee43R117 ?
If `lazy_relation_to_hash` can handle inventory objects as well as lazy inventory objects should it be renamed to something more general?

Reply:
Actually it looks like it is always setting the type to lazy, https://github.com/ManageIQ/manageiq/pull/17361/files#diff-7a3a5be928f76ef87ac9cee741b1ee43R57
Will this work with a regular inventory object?

Reply:
right, all links are trasfered to lazy_find, since that is the only one that is serializable

Reply:
right, I'll might merge it to 1 branch